src/chat.rs

```diff
@@ -604,6 +606,34 @@ impl ChatId {
         Ok(())
     }
 
+    pub(crate) async fn add_encrypted_msg(
+        self,
+        context: &Context,
+        timestamp_sort: i64,
+    ) -> Result<()> {
+        let text = stock_str::messages_e2e_encrypted(context).await;
+        add_info_msg_with_cmd(
+            context,
+            self,
+            &text,
+            SystemMessage::ChatE2ee,
+            timestamp_sort,
+            None,
+            None,
+            None,
+            None,
+        )
+        .await?;
+        Ok(())
+    }
+
+    /// Returns true if the chat is encrypted, plain chat.
+    pub(crate) async fn is_encrypted_and_not_special(self, context: &Context) -> Result<bool> {
+        let chat = Chat::load_from_db(context, self).await?;
+        let res = chat.is_encrypted(context).await? && !chat.is_device_talk();
+        Ok(res)
```

Comment:
Possible bug(blocking): Shouldn't this also check for `self>DC_CHAT_ID_LAST_SPECIAL && !chat.is_self_talk()`?